contracts/Staking.sol

```diff
@@ -137,44 +364,77 @@ contract Staking is AccessManagedUpgradeable, ReentrancyGuardUpgradeable, IStaki
-    }
-
+    /**
+     * @notice Updates the reward wallet beacon address
+     * @dev It's a reinitializer - used once only during contract deployment or upgrade from old version
+     * @param rewardWalletBeacon_ The address of the reward wallet beacon contract
+     */
     function updateRewardWalletBeacon(IBeacon rewardWalletBeacon_) external reinitializer(2) restricted override{
         require(address(rewardWalletBeacon_) != address(0), InvalidRewardWalletAddress());
         IBeacon oldBeacon = rewardWalletBeacon;
         rewardWalletBeacon = rewardWalletBeacon_;
         emit RewardWalletBeaconUpdated(oldBeacon, rewardWalletBeacon_);
     }
 
+    /**
+     * @notice Adds an address to the list of allowed fee receivers for the caller's node
+     * @dev Only callable by node owners
+     * @param receiver The address to add to the allowed receivers list
+     */
     function addAllowedReceiver(address receiver) external override {
         NodeId node = nodes.getNodeId(msg.sender);
         bool added = _nodesAllowedReceivers[node].add(receiver);
         require(added, ReceiverIsAlreadyAllowed(receiver));
         emit AllowedReceiverAdded(node, receiver);
     }
 
+    /**
+     * @notice Removes an address from the list of allowed fee receivers for the caller's node
+     * @dev Only callable by node owners
+     * @param receiver The address to remove from the allowed receivers list
+     */
     function removeAllowedReceiver(address receiver) external override {
         NodeId node = nodes.getNodeId(msg.sender);
         bool removed = _nodesAllowedReceivers[node].remove(receiver);
         require(removed, ReceiverWasNotAllowed(receiver));
         emit AllowedReceiverRemoved(node, receiver);
     }
 
+    /**
+     * @notice Requests all earned fees for a specific node
+     * @dev Only callable by node owners and allowed receivers
+     * @param node The node whose fees to request
+     */
     function requestAllFees(NodeId node) external override {
         requestFees(node, getEarnedFeeAmount(node));
     }
 
+    /**
+     * @notice Requests to send all earned fees to a specified address
+     * @dev Only callable by node owners
+     * @dev to address must be an allowed receiver if the list is not empty
+     * @param to The address to receive the fees
+     */
     function requestSendAllFees(address payable to) external override {
         requestSendFees(to, getEarnedFeeAmount(nodes.getNodeId(msg.sender)));
     }
 
+    /**
+     * @notice Sets the minimum self-stake requirement for nodes
+     * @dev Only callable by authorized addresses (restricted)
+     * @param amount The new self-stake requirement
+     */
     function setSelfStakeRequirement(Fair amount) external override restricted {
         selfStakeRequirement = amount;
         emit SelfStakeRequirementUpdated(amount);
     }
 
+    /**
+     * @notice Disables a node from receiving network rewards and removes it from the active pool
```

Comment:
"network stability rewards" because they can continue receiving block creation rewards from the network.